xarray/backends/api.py

```diff
@@ -837,6 +837,43 @@ def open_datatree(
     return backend.open_datatree(filename_or_obj, **kwargs)
```

Comment:
We _could_ have a default implementation here that calls `open_groups`, i.e.

```suggestion
    groups_dict = backend.open_datatree(filename_or_obj, **kwargs)
    return DataTree.from_dict(groups_dict)
```

The idea being that then backend developers don't actually have to implement `open_datatree` if they have implemented `open_groups`...

This was sort of discussed here (@keewis) https://github.com/pydata/xarray/pull/7437#discussion_r1279083937, but this seems like an rabbit hole that should be left for a future PR.

Reply:
not really, I was arguing that given any one of `open_dataarray`, `open_dataset` and `open_datatree` allows us to provide (somewhat inefficient) default implementations for the others. However, `open_groups` has a much closer relationship to `open_datatree`, so I think having a default implementation for `open_datatree` is fine (we just need to make sure that a backend that provides neither `open_groups` nor `open_datatree` doesn't complain about `open_groups` not existing if you called `open_datatree`).

So yeah, this might become a rabbit hole.

Reply:
Okay I see. That seems related, but also like a totally optional convenience feature that we should defer to later.